controllers/flinkcluster_submit_job_script.go

```diff
@@ -22,87 +22,136 @@ package controllers
+    local job_id
+
+    # Submit job and extract the job ID
+    echo "/opt/flink/bin/flink run $*" | tee -a submit_log
+    if /opt/flink/bin/flink run "$@" 2>&1 | tee -a submit_log; then
+        local -r job_id_indicator="Job has been submitted with JobID"
+        job_id=$(grep "${job_id_indicator}" submit_log | awk -F "${job_id_indicator}" '{printf $2}' | awk '{printf $1}')
+    fi
+
+    # Write result as YAML format to pod termination-log.
+    # On failure, write log only.
+    if [[ -z ${job_id+x} ]]; then
+        write_term_log "Failed to submit." "submit_log"
+        return 1
+    fi
+
+    # On success, write job ID and log.
+    echo "jobID: ${job_id}" >"${TERM_LOG}"
+    write_term_log "Successfully submitted!" "submit_log"
+
+    return 0
 }
 
-function submit_job() {
-	echo -e "\nSubmitting job..."
-	echo "/opt/flink/bin/flink run $@"
-	/opt/flink/bin/flink run "$@"
+function echo_log() {
+    local msg="$1"
+    local log_file="$2"
+    echo -e "${msg}" | tee -a "${log_file}"
 }
 
-function wait_for_job() {
-	while true; do
-		echo -e "\nWaiting for job to finish..."
-		list_jobs
-
-		# Find active job first.
-		# If the current job is restarted by the operator, there will be records of past stopped jobs.
-		# TODO: It needs to be improved to determine the job state with the submitted job id.
-		if list_jobs | grep -e "(SCHEDULED)" -e "(CREATED)" -e "(SUSPENDED)" \
-			-e "(RUNNING)" -e "(RESTARTING)" -e "(CANCELLING)" -e "(FAILING)" -e "(RECONCILING)"; then
-			echo -e "\nFound an active job."
-		else
-			if list_jobs | grep "(FINISHED)"; then
-				echo -e "\nJob has completed successfully, exiting 0"
-				return 0
-			fi
-			if list_jobs | grep "(FAILED)"; then
-				echo -e "\nJob failed, exiting 1"
-				return 1
-			fi
-			if list_jobs | grep "(CANCELED)"; then
-				echo -e "\nJob has been cancelled, exiting 2"
-				return 2
-			fi
-			if list_jobs | grep "No running jobs" && list_jobs | grep "No scheduled jobs"; then
-				echo -e "\nNo running and scheduled jobs, exiting 3"
-				return 3
-			fi
-			echo -e "\nUnknown job state, check it again in the next iteration."
-		fi
-		sleep 30
-	done
+# Write submit result to pod termination log file as YAML format.
+function write_term_log() {
+    local result_msg="$1"
+    local log_file="$2"
+
+    # Write result message.
+    echo "message: |" >>"${TERM_LOG}"
+    echo "  ${result_msg}" >>"${TERM_LOG}"
+
+    # Append submit log to message.
+    # Two space indentation is required to write strings in the form of YAML literal block scalar.
+    IFS=''
+    while read -r line; do
+        local term_log_size
+        local line_size
+
+        # Termination log is limited to 4KiB, so calculate written size so far.
+        term_log_size=$(stat -c %s "${TERM_LOG}")
+        line_size=${#line}
+        if ((term_log_size + line_size > 4096)); then
+            break
+        fi
+
+        # Insert indentation before printing line.
+        printf "  %s\n" "${line}" >>"${TERM_LOG}"
```

Comment:
I think it's more readable to move `>>"${TERM_LOG}"` out of this function, and rename this function to `format_log_message`.

Reply:
Fixed.